src/main/java/de/tum/in/www1/artemis/domain/Achievement.java

```diff
@@ -0,0 +1,166 @@
+public class Achievement implements Serializable {
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "title")
+    private String title;
+
+    @Column(name = "description")
+    private String description;
+
+    @Column(name = "icon")
+    private String icon;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "`rank`")
+    private AchievementRank rank;
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "`type`")
+    private AchievementType type;
+
+    @ManyToOne
+    private Course course;
+
+    @ManyToOne
+    private Exercise exercise;
+
+    @ManyToMany(mappedBy = "achievements")
+    @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
+    @JsonIgnoreProperties({ "achievements" })
+    private Set<User> users = new HashSet<>();
+
+    public Achievement() {
+    }
+
+    public Achievement(String title, String description, String icon, AchievementRank rank, AchievementType type, Course course, Exercise exercise) {
+        this.title = title;
+        this.description = description;
+        this.icon = icon;
+        this.rank = rank;
+        this.type = type;
+        this.course = course;
+        this.exercise = exercise;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(String title) {
+        this.title = title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public String getIcon() {
+        return icon;
+    }
+
+    public void setIcon(String icon) {
+        this.icon = icon;
+    }
+
+    public AchievementRank getRank() {
+        return rank;
+    }
+
+    public void setRank(AchievementRank rank) {
+        this.rank = rank;
+    }
+
+    public AchievementType getType() {
+        return type;
+    }
+
+    public void setType(AchievementType type) {
+        this.type = type;
+    }
+
+    public Set<User> getUsers() {
+        return users;
+    }
+
+    public void setUsers(Set<User> users) {
+        this.users = users;
+    }
+
+    public Course getCourse() {
+        return course;
+    }
+
+    public void setCourse(Course course) {
+        this.course = course;
+    }
+
+    public Exercise getExercise() {
+        return exercise;
+    }
+
+    public void setExercise(Exercise exercise) {
+        this.exercise = exercise;
+    }
+
+    @Override
+    public String toString() {
+        return "Achievement{" + "id=" + getId() + ", title='" + getTitle() + "'" + ", description='" + getDescription() + "'" + ", icon='" + getIcon() + "'" + ", rank='"
+                + getRank() + "'" + ", type='" + getType() + "'" + "}";
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I am not certain if this implementation of `equals` and `hashcode` is a good idea, as you are also calling equals and hashcode of other entities.  But I am not an expert so @krusche might help here

Maybe read through this here https://docs.jboss.org/hibernate/stable/core.old/reference/en/html/persistent-classes-equalshashcode.html.

Is title a natural key?

Reply:
Removing the functions for now as they are not needed in this PR